src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table = new Node[DEFAULT_INITIAL_CAPACITY];
+    private int threshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    private int size;
+
     @Override
     public void put(K key, V value) {
+        if (size == threshold) {
+            resize();
+        }
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+        } else {
+            Node<K, V> node = table[index];
+            while (node != null) {
+                if (Objects.equals(node.key, key)) {
+                    node.value = value;
+                    return;
+                } else if (node.next == null) {
+                    node.next = newNode;
+                    size++;
+                }
+                node = node.next;
+            }
+        }
+    }
 
+    private void resize() {
+        Node<K, V>[] oldTab = table;
+        table = new Node[oldTab.length * 2];
+        size = 0;
+        threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+        for (Node<K, V> node : oldTab) {
+            while (node != null) {
+                put(node.key, node.value);
+                node = node.next;
+            }
+        }
+    }
+
+    private int hash(Object key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
     }
 
     @Override
     public V getValue(K key) {
+        int index = hash(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
```

Comment:
You do not use Node<K, V> next.
You use this constructor once and set 3rd argument to null. So, why do you need 3rd argument?